src/core/raster/qgsrasterblock.cpp

```diff
@@ -872,3 +876,63 @@ QRect QgsRasterBlock::subRect( const QgsRectangle &extent, int width, int height
   QgsDebugMsgLevel( QStringLiteral( "subRect: %1 %2 %3 %4" ).arg( subRect.x() ).arg( subRect.y() ).arg( subRect.width() ).arg( subRect.height() ), 4 );
   return subRect;
 }
+
+bool QgsRasterBlock::minimum( double &minimum, int &row, int &column ) const
+{
+  if ( !mData )
+  {
+    minimum = std::numeric_limits<double>::quiet_NaN();
+    return false;
+  }
+
+  const std::size_t offset = qgis_gdal::min_element( mData, static_cast<std::size_t>( mWidth ) * static_cast< std::size_t>( mHeight ),
+                             QgsGdalUtils::gdalDataTypeFromQgisDataType( mDataType ), mHasNoDataValue, mNoDataValue );
+
+  row = static_cast< int >( offset / mWidth );
+  column = static_cast< int >( offset % mWidth );
+  minimum = value( offset );
+
+  return true;
+}
+
+bool QgsRasterBlock::maximum( double &maximum SIP_OUT, int &row SIP_OUT, int &column SIP_OUT ) const
+{
+  if ( !mData )
+  {
+    maximum = std::numeric_limits<double>::quiet_NaN();
+    return false;
+  }
+  const std::size_t offset = qgis_gdal::max_element( mData, static_cast<std::size_t>( mWidth ) * static_cast< std::size_t>( mHeight ),
+                             QgsGdalUtils::gdalDataTypeFromQgisDataType( mDataType ), mHasNoDataValue, mNoDataValue );
+
+  row = static_cast< int >( offset / mWidth );
+  column = static_cast< int >( offset % mWidth );
+  maximum = value( offset );
+
+  return true;
+}
+
+bool QgsRasterBlock::minimumMaximum( double &minimum, int &minimumRow, int &minimumColumn, double &maximum, int &maximumRow, int &maximumColumn ) const
+{
+  if ( !mData )
+  {
+    minimum = std::numeric_limits<double>::quiet_NaN();
+    maximum = std::numeric_limits<double>::quiet_NaN();
+    return false;
+  }
+
+  const auto offset = qgis_gdal::minmax_element( mData, static_cast<std::size_t>( mWidth ) * static_cast< std::size_t>( mHeight ),
+                      QgsGdalUtils::gdalDataTypeFromQgisDataType( mDataType ), mHasNoDataValue, mNoDataValue );
+  std::size_t minOffset = offset.first;
+  std::size_t maxOffset = offset.second;
```

Comment:
could be simplified as:
```suggestion
  const auto [minOffset, maxOffset] = qgis_gdal::minmax_element( mData, static_cast<std::size_t>( mWidth ) * static_cast< std::size_t>( mHeight ),
                      QgsGdalUtils::gdalDataTypeFromQgisDataType( mDataType ), mHasNoDataValue, mNoDataValue );
```